Feature/add new contribution gh 297

kdv24 opened the pull request.

kdv24 committed:
Add AddContribution file structure & routes

- needs more refactoring still

kdv24 committed:
Add AddContribution file structure & routes

- needs more refactoring still

kdv24 committed:
add initial formSections

kdv24 committed:
add initial formSections

kdv24 committed:
fix page location/routes & add formSections to index

kdv24 committed:
fix page location/routes & add formSections to index

kdv24 committed:
WIP- add values for dropdowns & textfields

kdv24 committed:
WIP- add values for dropdowns & textfields

kdv24 committed:
add test for AddContribution page

kdv24 committed:
add test for AddContribution page

kdv24 committed:
remove unused css, portal, and formModal

kdv24 committed:
remove unused css, portal, and formModal

kdv24 committed:
WIP - add AddContribution page to Contributions & update route

kdv24 committed:
WIP - add AddContribution page to Contributions & update route

andrewbiang888 commented on app/src/Pages/AddContribution/AddContribution.js:
```diff
@@ -0,0 +1,32 @@
+import React, { Component } from "react";
```
I think we should move this code into the app/src/Pages/Portal/Contributions/Contributions.js page (so combining the two pages).

andrewbiang888 replied:
Correction, it looks the pages you are creating should represent the path `/contributions/add` so maybe in that page. I would create the page here:
`/app/src/Pages/Portal/Contributions/Add` (we are trying to group the portal pages similar to how you would in Next.js)

kdv24 replied:
👍

andrewbiang888 commented on app/src/Pages/AddContribution/AddContribution.js:
```diff
@@ -0,0 +1,32 @@
+  }
+}
+
+export default connect(
```
When you combine the pages, this should be pulled out into `app/src/Pages/Portal/Contributions/index.js` and then do something like this:
```js
import { connect } from "react-redux";
import TopNavigation from "./Contributions";
import { isLoggedIn } from "../../state/ducks/auth";

export default connect(
state => { 
    return {state: state.auth}
  }, 
  dispatch => {
    return {
      login: (email,password) => dispatch(login(email,password)),
      dispatch
      }
    }
)(Contributions);

```

kdv24 replied:
Ok, assuming `TopNavigation` should be `Contributions`.

andrewbiang888 reviewed (commented):
Making good progress. Once I finish the Manage Portal section I will be able help with the `/contributions` tables!

jaronheard commented on app/src/components/Forms/AddContribution/index.js:
```diff
@@ -0,0 +1,56 @@
+          {formSections.amountOfContribution} 
+          {formSections.oaeContributionType} 
+          {formSections.paymentMethod} 
+          {formSections.checkNumber} 
```
I was imagining that the formSections would be a larger section, e.g. all of the above would be in `formSections.basic`. Then could be wrapped in a `div` that handles the layout of all of the fields within that section.

kdv24 replied:
👍

kdv24 replied:
@jaronheard right now I have the  `header`, `basic`, `contributor`, and `other details` sections broken out into separate forms that are all pulled together to create the AddContribution page. I did it that way so I could break it all apart in my mind as I was trying understand the structure of the code, but one flaw as it sits right now is the `submit` button is probably only aware of the `header` form where it lives.  😬 

So, before I do any more that may be outside of what you're describing here, maybe we can talk tomorrow.  I'd also love to chat with you or maybe @andrewbiang888 about some styling questions.

kdv24 commented:
Thanks for the comments! Obviously I haven't done anything with styling at this point, but I have questions around that when I get there too.  I'm looking forward to seeing the Manage Portal section.

kdv24 committed:
add header section

kdv24 committed:
add header section

kdv24 committed:
break basics section out into own

kdv24 committed:
break basics section out into own

kdv24 committed:
add contributor section

kdv24 committed:
add contributor section

kdv24 committed:
updated AddContribution file structure in pages

kdv24 committed:
updated AddContribution file structure in pages

kdv24 committed:
remove initial AddContributionForm

kdv24 committed:
remove initial AddContributionForm

kdv24 committed:
add fields to contributor section

kdv24 committed:
add fields to contributor section

kdv24 committed:
add "other details" section and fields

kdv24 committed:
add "other details" section and fields

kdv24 committed:
rename header section & add test

kdv24 committed:
rename header section & add test

kdv24 committed:
add sections from "other details" per ORESTAR categories google doc

kdv24 committed:
add sections from "other details" per ORESTAR categories google doc

kdv24 committed:
removed unused code & change initial values to "" for now

kdv24 committed:
removed unused code & change initial values to "" for now

kdv24 committed:
removed unused code & change initial values to "" for now

kdv24 committed:
removed unused code & change initial values to "" for now

kdv24 committed:
move all AddContribution components into one with single index & delete old components

kdv24 committed:
move all AddContribution components into one with single index & delete old components

kdv24 committed:
add in missing initial values

kdv24 committed:
add in missing initial values

kdv24 committed:
change invoice header to non-form & fix otherDetails section typo

kdv24 committed:
change invoice header to non-form & fix otherDetails section typo

kdv24 committed:
add more validation and clean up of fields

kdv24 committed:
add more validation and clean up of fields

kdv24 committed:
add minimal styling attempt

kdv24 committed:
add minimal styling attempt

jaronheard committed:
Added formFields to Form components

In addition to formSections, the Form component now returns a
formSections object, so that you can layout and style fields
individually.

jaronheard committed:
Example use of formFields

Design asked you to make the email field hot pink. Wyd?

jaronheard committed:
Revert example use of formFields

kdv24 committed:
WIP - start adding styles

kdv24 committed:
WIP- add initial flexbox styling

kdv24 committed:
add FormControl fullWidth prop

kdv24 committed:
WIP- more styling

kdv24 committed:
WIP- styles

kdv24 committed:
WIP- initial flexbox styling with issues, including mobile

kdv24 committed:
WIP - switch to CSS Grid for better responsiveness/mobile

- looks pretty good, but still some weird line spacing issues to fix

kdv24 committed:
WIP - switch to CSS Grid for better responsiveness/mobile

- looks pretty good, but still some weird line spacing issues to fix
- need to handle Employer State & Zip when narrow for mobile, esp w/o default value

kdv24 committed:
fix merge conflict

kdv24 committed:
begin work on ContributionSubmitted - add initial file structure

kdv24 committed:
WIP - convert form values to not need ContributionSubmittedForm

- add initial styles
- disconnect from form
- add initial mock values

kdv24 committed:
WIP - more styling tweaks

- will benefit from refactoring when it's set

punkael committed:
add DateField component (#392)

kdv24 committed:
convert ContributionSubmitted to a form

kdv24 committed:
create new field without underline for displaying values

kdv24 committed:
add date field to form

kdv24 committed:
start work on NewExpenses

- set up initial folder structure and paths

kdv24 committed:
add fields and yup for expenditures

kdv24 committed:
add expense route

kdv24 committed:
update Portal to include Expenses

- for some reason I can't get my config to quit changing single quotes to double
- so it looks like a ton of changes, but is only the import, url, & route

kdv24 committed:
move Occupation Letter Date to employer section

kdv24 committed:
update button text & add wrapper

- colors and some styling of buttons not working yet

kdv24 committed:
refactor header & styling & update content

kdv24 committed:
clearn up with refactored format from AddExpense

kdv24 committed:
fix header row length

kdv24 committed:
clearn up with refactored format from AddExpense

kdv24 committed:
add ExpenseDetail form, index, page, routes, and tests

kdv24 committed:
add button responsiveness kinda

- needs more once we know desired look on mobile/smaller screen

kdv24 committed:
add better folder structure for contribution pages

kdv24 committed:
add ContributionReady form, index, page, test, routes

kdv24 committed:
add ExpensesDetail pages, form, routes, tests, index

kdv24 committed:
add ContributionNeedsReview city view

- adjusted and added directory structure for city views
- added Activity Stream component for adding to bottom of city views
- Activity Stream has mock data in it now, but I don't know actual shape

kdv24 committed:
update header - missing icon/dropdown

kdv24 committed:
fix merge conflicts

Commit:
fixing file extension

Commit:
fixing file extension

Commit:
fixing snapshot tests
